src/crimson/common/config_proxy.h

```diff
@@ -45,22 +45,38 @@ class ConfigProxy : public seastar::peering_sharded_service<ConfigProxy>
       new_values->changed.clear();
       func(*new_values);
 
+      ObserverMgr<ConfigObserver>::rev_obs_map_t rev_obs;
+
       // always apply the new settings synchronously on the owner shard, to
       // avoid racings with other do_change() calls in parallel.
       owner.values.reset(new_values);
-      owner.obs_mgr.apply_changes(owner.values->changed,
-                                  owner, nullptr);
+      owner.obs_mgr.gather_changes(owner.values->changed, owner,
+                                   [&rev_obs](ConfigObserver *obs,
+                                              const std::string &key) {
+                                     rev_obs[obs].insert(key);
+                                   }, nullptr);
+      for (auto& [obs, keys] : rev_obs) {
+        obs->handle_conf_change(owner, keys);
+      }
+
+      rev_obs.clear();
 
       return seastar::parallel_for_each(boost::irange(1u, seastar::smp::count),
-                                        [&owner, new_values] (auto cpu) {
+                                        [&owner, new_values, &rev_obs] (auto cpu) {
```

Comment:
`rev_obs` is allocated on the stack. and it can not be passed around using continuations by capturing its reference. instead of reusing this `rev_obs`, i'd suggest move it to the lambda in `invoke_on()` call.